api/elastic.ts

```diff
@@ -0,0 +1,92 @@
+import {Client} from '@elastic/elasticsearch'
+import {snapshots} from './handler'
+import {chain, left, right, TaskEither, tryCatch} from 'fp-ts/lib/TaskEither'
+import {ApplicationError, StatusCodes} from './http'
+import {Snapshot} from './model'
+import {pipe} from 'fp-ts/lib/pipeable'
+
+export const getClient = (): Client => {
+    return new Client({node: process.env.ELASTIC_SEARCH})
+}
+
+const getByHash = (index: string, hash: string) => (es: Client) =>
+    es.search({
+        index: index,
+        body: {
+            query: {
+                match: {
+                    hash: {
+                        query: hash,
+                    },
+                }
+            }
+        }
+    })
+
+
+const getByHeight = (index: string, height: string) => (es: Client) =>
+    es.search({
+        index: index,
+        body: {
+            size: -1,
```

Comment:
You are fetching only one snapshot here, as height is a unique value. The same applies for `getByHash` function.
That being said, it should be `size: 1`